lib/reply.js

```diff
@@ -512,10 +513,9 @@ function handleError (reply, error, cb) {
     } else if (error.statusCode >= 400) {
       statusCode = error.statusCode
     }
+    res.statusCode = statusCode
```

Comment:
Can you avoid duplication of this block and the one you added in the default error handler by placing them in a new `.js` file? Thanks.